openedx/core/djangoapps/theming/core.py

```diff
@@ -1,62 +1,32 @@
 """
 Core logic for Comprehensive Theming.
 """
-from path import Path
-
+import os.path
+from path import Path as path
 from django.conf import settings
 
+from .helpers import (
+    get_project_root_name,
+)
 
-def comprehensive_theme_changes(theme_dir):
-    """
-    Calculate the set of changes needed to enable a comprehensive theme.
-
-    Arguments:
-        theme_dir (path.path): the full path to the theming directory to use.
-
-    Returns:
-        A dict indicating the changes to make:
-
-            * 'settings': a dictionary of settings names and their new values.
-
-            * 'template_paths': a list of directories to prepend to template
-                lookup path.
 
-    """
-
-    changes = {
-        'settings': {},
-        'template_paths': [],
-    }
-    root = Path(settings.PROJECT_ROOT)
-    if root.name == "":
-        root = root.parent
-
-    component_dir = theme_dir / root.name
-
-    templates_dir = component_dir / "templates"
-    if templates_dir.isdir():
-        changes['template_paths'].append(templates_dir)
-
-    staticfiles_dir = component_dir / "static"
-    if staticfiles_dir.isdir():
-        changes['settings']['STATICFILES_DIRS'] = [staticfiles_dir] + settings.STATICFILES_DIRS
-
-    locale_dir = component_dir / "conf" / "locale"
-    if locale_dir.isdir():
-        changes['settings']['LOCALE_PATHS'] = [locale_dir] + settings.LOCALE_PATHS
-
-    return changes
-
-
-def enable_comprehensive_theme(theme_dir):
+def enable_comprehensive_theme(themes_dir):
     """
     Add directories to relevant paths for comprehensive theming.
+    :param themes_dir: path to base theme directory
```

Comment:
Could we generalize this so that it can pick up theme directories from more than one location? In an ideal world, there would be a Stevedore plugin extension so that any Python package can register a new theme, and then this code could iterate all found theme plugins. Is this planned for a later story?